java-frontend/src/main/java/org/sonar/java/bytecode/cfg/BytecodeCFGBuilder.java

```diff
@@ -315,10 +316,10 @@ public void visitTypeInsn(int opcode, String type) {
     public void visitJumpInsn(int opcode, Label label) {
       if(opcode == GOTO || opcode == JSR) {
         currentBlock.terminator = new Instruction(opcode);
-        blockByLabel.computeIfAbsent(label, l -> currentBlock.createSuccessor());
+        currentBlock.successors = Collections.singletonList(blockByLabel.computeIfAbsent(label, l -> currentBlock.createSuccessor()));
         return;
       }
-      blockByLabel.computeIfAbsent(label, l -> currentBlock.createTrueSuccessor());
+      currentBlock.trueBlock = blockByLabel.computeIfAbsent(label, l -> currentBlock.createTrueSuccessor());
```

Comment:
Same as above, you can set `trueBlock` with result of `createSuccessor` taking label as argument, and this will actually eliminate the need for dedicated method